src/solc_0.8/defi/rules/RequirementsRules.sol

```diff
@@ -0,0 +1,280 @@
+
+    function getERC1155ListRequirement(address contractERC1155)
+        external
+        view
+        _isContract(contractERC1155)
+        returns (RequirementRuleERC1155 memory)
+    {
+        require(_islistERC1155Member(IERC1155(contractERC1155)), "RequirementsRules: contract is not in the list");
+        return _listERC1155[IERC1155(contractERC1155)];
+    }
+
+    function deleteERC721fromListRequirement(IERC721 contractERC721)
+        external
+        onlyOwner
+        _isContract(address(contractERC721))
+    {
+        require(_islistERC721Member(contractERC721), "RequirementsRules: contract is not in the list");
+        uint256 indexToDelete = _listERC721[contractERC721].index;
+        IERC721 addrToMove = _listERC721Index[_listERC721Index.length - 1];
+        _listERC721Index[indexToDelete] = addrToMove;
+        _listERC721[addrToMove].index = indexToDelete;
+        _listERC721Index.pop();
+
+        emit ERC721RequirementDeleted(address(contractERC721));
+    }
+
+    function deleteERC1155fromListRequirement(IERC1155 contractERC1155)
+        external
+        onlyOwner
+        _isContract(address(contractERC1155))
+    {
+        require(_islistERC1155Member(contractERC1155), "RequirementsRules: contract is not in the list");
+        uint256 indexToDelete = _listERC1155[contractERC1155].index;
+        IERC1155 addrToMove = _listERC1155Index[_listERC1155Index.length - 1];
+        _listERC1155Index[indexToDelete] = addrToMove;
+        _listERC1155[addrToMove].index = indexToDelete;
+        _listERC1155Index.pop();
+
+        emit ERC1155RequirementDeleted(address(contractERC1155));
+    }
+
+    function _islistERC721Member(IERC721 reqContract) internal view returns (bool) {
+        if (_listERC721Index.length == 0) return false;
+
+        return (_listERC721Index[_listERC721[reqContract].index] == reqContract);
+    }
+
+    function _islistERC1155Member(IERC1155 reqContract) internal view returns (bool) {
+        if (_listERC1155Index.length == 0) return false;
+
+        return (_listERC1155Index[_listERC1155[reqContract].index] == reqContract);
+    }
+
+    function _checkERC721Requirements(address account) internal view returns (uint256) {
+        uint256 _maxStake = 0;
+        for (uint256 i = 0; i < _listERC721Index.length; i++) {
+            uint256 balanceOf = 0;
+            uint256 balanceOfId = 0;
+            IERC721 reqContract = _listERC721Index[i];
+
+            if (_listERC721[reqContract].balanceOf == true) {
+                balanceOf = reqContract.balanceOf(account);
+            }
+
+            for (uint256 j = 0; j < _listERC721[reqContract].ids.length; j++) {
+                address owner = reqContract.ownerOf(_listERC721[reqContract].ids[j]);
+                if (owner == account) {
+                    ++balanceOfId;
+                }
+            }
+
+            if (balanceOf < _listERC721[reqContract].minAmountBalanceOf) {
+                balanceOf = 0;
+            }
```

Comment:
this won't work.... we said that we can't use max stake to check the min amount

if you meet the min amount requirement for ERC721 but don't for ERC1155, you should NOT be able to stake
Here if you have maxStake > 0 for ERC721 and maxStake = 0, you'll be able to stake

Reply:
done -> https://github.com/thesandboxgame/sandbox-smart-contracts/pull/605/commits/7fb47d3f89eb9851a8b05f6b1b654433fc6ff2fc